### DISCUSSION TITLE
[Question]: I cannot figure out if Internet access works in librechat?

### DISCUSSION THREAD

**Discussion by virtuman** [danny-avila/LibreChat#1923] [Q&A]
### What is your question?

Hello, can you please explain if librechat has functionality to scrape web pages or is it on the roadmap?

### More Details

scrape webpages would be nice to have. chatgpt does it already. maybe Google search is what i should be looking at? 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by fuegovic** [comment]
We have a couple of plugins that can access the internet, for both the plugins endpoint and the assistants endpoint.

**Reply by virtuman** [reply]
could you elaborate on how to use it? and whether i need to spin up any headless browsers or something on my infrastructure to improve performance and accessibility of javascript-based websites?

**Reply by danny-avila** [reply]
none of the plugins do any complex scraping like that. some do proprietary methods of search engine retrieval, like Tavily

I would like to set my sights on this when I am implementing RAG for the project.

**Reply by myroslav** [reply]
@danny-avila, have you considered [Browser-Use](https://browser-use.com/)? It is available under MIT license for [local setup](https://docs.browser-use.com/development/local-setup). It can "see" the page (screenshot), if vision is enabled. For "easy start" [hosted (cloud) option](https://cloud.browser-use.com/) is available.

BTW there is already [MCP server](https://github.com/Saik0s/mcp-browser-use) that wraps [browser-use/web-ui](https://github.com/browser-use/web-ui).
